### ISSUE TITLE
Create DocumentDB never completes 

### ISSUE THREAD

**Issue by johnstairs** [Azure/azure-sdk-for-net#3541]
This code consistently hangs for me:

```
Task<IDocumentDBAccount> task = azure.DocumentDBAccounts
    .Define("mydb123")
    .WithRegion(Region.USWest)
    .WithExistingResourceGroup("myrg123)
    .WithKind(DatabaseAccountKind.GlobalDocumentDB)
    .WithEventualConsistency()
    .WithWriteReplication(Region.USWest)
    .CreateAsync();

IDocumentDBAccount taskResult = task.Result;
```

The Cosmos DB account actually gets created but the task never completes.

**Comment by johnstairs** [comment]
I don't think this is the same issue. This is a simple console app using .net 4.5.2 and 4.6. Fiddler shows repeated requests to check the status, so I don't think this is a true synchronization/deadlock problem. I could be wrong though :)

**Comment by hovsepm** [comment]
@johnstairs Could you upgrade your project reference to Fluent 1.2 and let me know it it still hangs?

**Comment by hovsepm** [comment]
@johnstairs DocumentDB is now CosmosDB

**Comment by martinsawicki** [comment]
addressed in v1.3
